module/VuFind/src/VuFind/Auth/OpenIDConnect.php

```diff
@@ -360,6 +364,50 @@ public function getSessionInitiator(string $target): ?string
         return $this->getProvider()->authorization_endpoint . '?' . http_build_query($params);
     }
 
+    /**
+     * Perform cleanup at logout time.
+     *
+     * @param string $url URL to redirect user to after logging out.
+     *
+     * @return string Redirect URL (modified for OpenIDConnect logout).
+     */
+    public function logout($url)
+    {
+        $redirectUrl = $url;
+        $end_session_endpoint = false;
+
+        $logout = $this->getConfig('logout');
+
+        if (!$logout) {
+            // No logout configured, so don't logout from service provider.
+            $this->debug('no logout URL given');
+        } elseif (filter_var($logout, FILTER_VALIDATE_URL)) {
+            // A valid URL was configured, use it.
+            $end_session_endpoint = $logout;
+        } else {
+            // Get end_session_endpoint from provider.
+            $provider = $this->getProvider();
+            $end_session_endpoint = $provider->end_session_endpoint ?? null;
+        }
+
+        if ($end_session_endpoint) {
+            // Retrieve id_token from session.
+            $idToken = $this->session->oidc_id_token ?? null;
+            if ($idToken === null) {
+                $this->logWarning('No id_token found in session data');
+            } else {
+                $params = [
+                    'id_token_hint' => $idToken,
+                    'post_logout_redirect_uri' => $url,
+                ];
+                $redirectUrl = $end_session_endpoint . '?' . http_build_query($params);
```

Comment:
I think it's not safe to assume that there isn't already a question mark in the endpoint URL.
```suggestion
                $redirectUrl = $end_session_endpoint . (str_contains($end_session_endpoint, '?') ? '&' : '?') . http_build_query($params);
```

Reply:
Thanks, I made the code more robust here as you suggested. There remains similar code for the authorization_endpoint which I did not change (can be done if there is really a problem).

Reply:
No need for now since it's not in scope of this PR. :)